rust-src/concordium_base/src/protocol_level_tokens/token_module_state.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    common::{
+        cbor,
+        cbor::{Bytes, CborSerializationResult, SerializationOptions, UnknownMapKeys},
+    },
+    protocol_level_tokens::RawCbor,
+};
+
+use concordium_base_derive::{CborDeserialize, CborSerialize};
+
+/// Protocol level token (PLT) module state
+#[derive(Debug, Clone, Eq, PartialEq, CborSerialize, CborDeserialize)]
+pub struct TokenModuleState {
+    /// The name of the token
+    pub name:       String,
+    // /// A URL pointing to the token metadata
+    // TODO comment in as part of COR-1385
+    // pub metadata:   MetadataUrl,
+    /// Whether the token supports an allow list.
+    pub allow_list: Option<bool>,
+    /// Whether the token supports a deny list.
+    pub deny_list:  Option<bool>,
+    /// Whether the token is mintable.
+    pub mintable:   Option<bool>,
+    /// Whether the token is burnable.
+    pub burnable:   Option<bool>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, CborSerialize, CborDeserialize)]
+pub struct MetadataUrl {
```

Comment:
Also, should we move this into it's own file in line with how the other shared types? This will also be part of the token initialization parameters.